arangod/Cluster/MaintenanceFeature.cpp

```diff
@@ -195,9 +189,10 @@ Result MaintenanceFeature::addAction(
     // similar action not in the queue (or at least no longer viable)
     if (curAction == nullptr || curAction->done()) {
 
-      createAction(newAction, executeNow);
-
-      if (!newAction || !newAction->ok()) {
+      if (newAction && newAction->ok()) {
```

Comment:
I approve of this change.  Just know that the two addAction() routines are going to be rewritten in my next checkin.  I will incorporate most of this change.  The rewrite has different goals, it is not a criticism of this change.